File: 2108-find-first-palindromic-string-in-the-array.py
"""
2108. Find First Palindromic String in the Array

Submitted: December 4, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 17.6 MB (beats 6.80%)
"""

class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            if word == word[::-1]: return word
        return ""

File: 2108-find-first-palindromic-string-in-the-array.cpp
/*
2108. Find First Palindromic String in the Array

Submitted: October 28, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 23.63 MB (beats 82.86%)
*/

class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        for (const string& word : words) {
            bool isPalindrome = true;
            string::const_iterator forward;
            string::const_reverse_iterator backward;
            for (forward = word.cbegin(), backward = word.crbegin(); forward != word.end(); ++forward, ++backward) {
                if (*forward != *backward) {
                    isPalindrome = false;
                    break;
                }
            }
            if (!isPalindrome) continue;
            return word;
        }
        return "";
    }
};
